pkgs/development/python-modules/beatprints/default.nix

```diff
@@ -0,0 +1,65 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchPypi,
+  pillow,
+  buildPythonPackage,
+  poetry-core,
+  requests,
+  pylette,
+  lrclibapi,
+  fonttools,
+  questionary,
+  rich,
+  toml,
+}:
+let
+  pillow_10_4_0 = pillow.overridePythonAttrs (old: rec {
+    version = "10.4.0";
+    src = fetchPypi {
+      inherit version;
+      pname = "pillow";
+      hash = "sha256-Fmwc1NJDCbMNYfefSpEUt7IxPXRQkSJ3hV/139fNSgY=";
+    };
+  });
+in
+buildPythonPackage rec {
+  pname = "BeatPrints";
+  version = "1.1.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "TrueMyst";
+    repo = "BeatPrints";
+    rev = "v${version}";
+    hash = "sha256-KDMt+B3tyh/20N2We82ya90xZJAhCkvbdnTgZDw1GYg=";
+  };
+
+  build-system = [
+    poetry-core
+  ];
+
+  dependencies = [
+    requests
+    pylette
+    lrclibapi
+    fonttools
+    questionary
+    rich
+    toml
+    pillow_10_4_0
```

Comment:
We can't accept the propagation of a versioned package since it causes a PYTHONPATH conflict.
Is this package incompatible with pillow>=11?

Reply:
It is indeed required by the dependencies listed in the project for the [current release](https://github.com/TrueMyst/BeatPrints/blob/v1.1.3/pyproject.toml#L18)

Reply:
If someone tries to install this package and other packages dependent on "pillow" at the same time, they will conflict.
We cannot accept it as a library if it does not work with pillow>=11.

Reply:
Wokey, I did the same as the required PR #373544 . I tried to use `substituteInPlace` for the `pillow` version. It seems to work well (CLI and lib).